apollo-federation/src/connectors/json_selection/methods/public/or.rs

```diff
@@ -0,0 +1,317 @@
+use apollo_compiler::collections::IndexMap;
+use serde_json_bytes::Value as JSON;
+use shape::Shape;
+use shape::location::SourceId;
+
+use crate::connectors::json_selection::ApplyToError;
+use crate::connectors::json_selection::ApplyToInternal;
+use crate::connectors::json_selection::MethodArgs;
+use crate::connectors::json_selection::VarsWithPathsMap;
+use crate::connectors::json_selection::immutable::InputPath;
+use crate::connectors::json_selection::location::Ranged;
+use crate::connectors::json_selection::location::WithRange;
+use crate::impl_arrow_method;
+
+impl_arrow_method!(OrMethod, or_method, or_shape);
+/// Given 2 or more values to compare, returns true if any of the values are truthy or false if none of them are truthy.
+///
+/// Examples:
+/// $(true)->or(false)            results in true
+/// $(false)->or(true)            results in true
+/// $(true)->or(true)             results in true
+/// $(false)->or(false)           results in false
+/// $(false)->or(false, true)     results in true
+fn or_method(
+    method_name: &WithRange<String>,
+    method_args: Option<&MethodArgs>,
+    data: &JSON,
+    vars: &VarsWithPathsMap,
+    input_path: &InputPath<JSON>,
+) -> (Option<JSON>, Vec<ApplyToError>) {
+    let mut result = match data {
+        JSON::Bool(value) => *value,
+        _ => {
+            return (
+                None,
+                vec![ApplyToError::new(
+                    format!(
+                        "Method ->{} can only be applied to boolean values.",
+                        method_name.as_ref()
+                    ),
+                    input_path.to_vec(),
+                    method_name.range(),
+                )],
+            );
+        }
+    };
+
+    if method_args.and_then(|args| args.args.first()).is_none() {
+        return (
+            None,
+            vec![ApplyToError::new(
+                format!("Method ->{} requires arguments", method_name.as_ref()),
+                input_path.to_vec(),
+                method_name.range(),
+            )],
+        );
+    };
```

Comment:
```suggestion
   let Some(MethodArgs { args, .. }) = method_args else {
        return (
            None,
            vec![ApplyToError::new(
                format!("Method ->{} requires arguments", method_name.as_ref()),
                input_path.to_vec(),
                method_name.range(),
            )],
        );
    };
```

Lets us unnest the loop later. Doesn't check that there's at least one argument, but do we really care at runtime? We'll start statically checking this in this release anyway, so we could save some work/code 🤷